src/System.IO.Compression.Brotli/System/IO/Compression/BrotliStream.cs

```diff
@@ -229,16 +207,29 @@ private void EnsureNotDisposed()
                 throw new ObjectDisposedException(BrotliEx.StreamDisposed);
         }
 
+        internal void RemoveBytes(int numberOfBytes)
+        {
+            ArraySegment<byte> buf;
+            if (_bufferStream.TryGetBuffer(out buf))
+            {
+                Buffer.BlockCopy(buf.Array, numberOfBytes, buf.Array, 0, (int)_bufferStream.Length - numberOfBytes);
+                _bufferStream.SetLength(BufferStream.Length - numberOfBytes);
+            }
+            else
+            {
+                throw new UnauthorizedAccessException();
```

Comment:
Because it means something wrong with memory allocation or smth else now. I think it is not necessary  exception, but useful